sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -206,15 +206,20 @@ public virtual TokenCredential Credential
                 }
                 else
                 {
-                    _credential = new ClientSecretCredential(
-                        GetVariable("TENANT_ID"),
-                        GetVariable("CLIENT_ID"),
-                        GetVariable("CLIENT_SECRET"),
-                        new ClientSecretCredentialOptions()
-                        {
-                             AuthorityHost = new Uri(AuthorityHostUrl)
-                        }
-                    );
+                    var clientSecret = ClientSecret;
+                    if (string.IsNullOrWhiteSpace(clientSecret))
+                    {
+                        _credential = new DefaultAzureCredential();
+                    }
+                    else
+                    {
+                        _credential = new ClientSecretCredential(
+                            TenantId,
+                            ClientId,
```

Comment:
Will `ClientId` being recorded cause any issues if someone else is running the tests that you recorded?

Reply:
no this is just for live/record mode, not playback.